cdap-data-fabric/src/main/java/co/cask/cdap/store/NamespaceStore.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.store;
+
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.NamespaceMeta;
+
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Store for namespaces.
+ */
+public interface NamespaceStore {
```

Comment:
I remember there was a jira to have the store throw appropriate exceptions instead of returning nulls. Is this a good time to implement that, just for ``NamespaceStore`` to begin with?